src/python/pants/backend/helm/subsystems/k8s_parser.py

```diff
@@ -0,0 +1,150 @@
+from pants.engine.engine_aware import EngineAwareParameter
+from pants.engine.fs import CreateDigest, Digest, FileContent
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.util.docutil import git_url
+from pants.util.logging import LogLevel
+from pants.util.strutil import softwrap
+
+logger = logging.getLogger(__name__)
+
+_HELM_K8S_PARSER_SOURCE = "k8s_parser_main.py"
+_HELM_K8S_PARSER_PACKAGE = "pants.backend.helm.subsystems"
+
+
+class HelmKubeParserSubsystem(PythonToolRequirementsBase):
+    options_scope = "helm-k8s-parser"
+    help = "Used perform modifications to the final output produced by Helm charts when they've been fully rendered."
+
+    default_version = "hikaru==0.11.0b"
+
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.7,<3.10"]
+
+    register_lockfile = True
+    default_lockfile_resource = (_HELM_K8S_PARSER_PACKAGE, "k8s_parser.lock")
+    default_lockfile_path = (
+        f"src/python/{_HELM_K8S_PARSER_PACKAGE.replace('.', '/')}/k8s_parser.lock"
+    )
+    default_lockfile_url = git_url(default_lockfile_path)
+
+
+class HelmKubeParserLockfileSentinel(GenerateToolLockfileSentinel):
+    resolve_name = HelmKubeParserSubsystem.options_scope
+
+
+@rule
+def setup_k8s_parser_lockfile_request(
+    _: HelmKubeParserLockfileSentinel,
+    post_renderer: HelmKubeParserSubsystem,
+    python_setup: PythonSetup,
+) -> GeneratePythonLockfile:
+    return GeneratePythonLockfile.from_tool(
+        post_renderer, use_pex=python_setup.generate_lockfiles_with_pex
+    )
+
+
+@dataclass(frozen=True)
+class _HelmKubeParserTool:
+    pex: VenvPex
+
+
+@rule
+async def build_k8s_parser_tool(k8s_parser: HelmKubeParserSubsystem) -> _HelmKubeParserTool:
+    parser_sources = pkgutil.get_data(_HELM_K8S_PARSER_PACKAGE, _HELM_K8S_PARSER_SOURCE)
+    if not parser_sources:
+        raise ValueError(
+            f"Unable to find source to {_HELM_K8S_PARSER_SOURCE!r} in {_HELM_K8S_PARSER_PACKAGE}"
+        )
+
+    parser_file_content = FileContent(
+        path="__k8s_parser.py", content=parser_sources, is_executable=True
+    )
+    parser_digest = await Get(Digest, CreateDigest([parser_file_content]))
+
+    parser_pex = await Get(
+        VenvPex,
+        PexRequest,
+        k8s_parser.to_pex_request(
+            main=EntryPoint(PurePath(parser_file_content.path).stem), sources=parser_digest
+        ),
+    )
+    return _HelmKubeParserTool(parser_pex)
+
+
+@dataclass(frozen=True)
+class ParseKubeManifestRequest(EngineAwareParameter):
+    filename: str
+    digest: Digest
+
+    def debug_hint(self) -> str | None:
+        return self.filename
+
+
+@dataclass(frozen=True)
+class ParsedKubeManifest:
+    filename: str
+    found_image_refs: tuple[tuple[int, YamlPath, str], ...]
+
+
+@rule(desc="Parse Kubernetes resource manifest", level=LogLevel.DEBUG)
```

Comment:
Ditto allowing the `level` to default on `@rules` which wrap processes with better descriptions.